File: code/library/src/main/java/org/fbme/lib/iec61499/declarations/FBTypeReference.java
package org.fbme.lib.iec61499.declarations;

import org.fbme.lib.common.ContainedElement;
import org.fbme.lib.common.Reference;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface FBTypeReference extends ContainedElement {

    @NotNull Reference<FBTypeDeclaration> getFBType();

    @Nullable ResourceTypeDeclaration getContainer();
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/FBTypeReference.kt
package org.fbme.lib.iec61499.declarations

import org.fbme.lib.common.ContainedElement
import org.fbme.lib.common.Reference

interface FBTypeReference : ContainedElement {
    val fBType: Reference<FBTypeDeclaration>
    override val container: ResourceTypeDeclaration?
}
